### PR TITLE
update `time` dependency in build-script

### PR DESCRIPTION
**Pull request by syphar**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -4016,6 +4026,16 @@ dependencies = [
...
+[[package]]
+name = "time"
+version = "0.3.6"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "c8d54b9298e05179c335de2b9645d061255bcd5155f843b3e328d2cfe0a5b413"
+dependencies = [
+ "libc",
+ "num_threads",
+]
+
```
Comment (jyn514) [inline Cargo.lock:4038]: not a giant fan that this introduces a new duplicate dependency - what else depends on time 0.1?

Comment (Nemo157) [reply Cargo.lock:4038]:
```
time v0.1.43
├── chrono v0.4.19 (*)
├── docs-rs v0.6.0 (/home/nemo157/sources/docs.rs)
├── hyper v0.10.16 (*)
└── zip v0.5.13 (*)

time v0.2.27
└── rusoto_signature v0.46.0 (*)

time v0.3.6
[build-dependencies]
└── docs-rs v0.6.0 (/home/nemo157/sources/docs.rs)
```

For one thing, we do as a non-build dependency for `iron` compatibility :sweat_smile:

Comment (syphar) [reply Cargo.lock:4038]:
```
$ cargo tree -i time:0.1.43 --depth 1        

time v0.1.43
├── chrono v0.4.19
├── docs-rs v0.6.0 (/Volumes/DATA/src/rust-lang/docs.rs)
├── hyper v0.10.16
└── zip v0.5.13
[build-dependencies]
└── docs-rs v0.6.0 (/Volumes/DATA/src/rust-lang/docs.rs)
```

also 

```
$ cargo tree -i time:0.2.27 --depth 1             
                                                                                                                                   
time v0.2.27
└── rusoto_signature v0.46.0
```

Comment (Nemo157) [reply Cargo.lock:4038]: Time's such a tiny crate that I don't think duplicating it matters, it should eventually coalesce as other crates upgrade (and we get rid of `iron`) too.

Comment (jyn514) [reply Cargo.lock:4038]: Could we make the build-dependency time 0.2, then? Or upgrade the 0.2 version to 0.3? I don't think we need 3 seperate versions ...

Comment (Nemo157) [reply Cargo.lock:4038]: As soon as we upgrade `sentry` we'll have 0.3 introduced either way.

Comment (syphar) [reply Cargo.lock:4038]: Yeah, everything around `time` is kind of a mess currently. 

But tricky to upgrade packages without using these 3 different versions, and personally I think this is not reason enough not to upgrade other packages, right? 

We could revisit the question when I upgraded all packages where possible, and see which versions remain.

Comment (syphar) [reply Cargo.lock:4038]: just checked, the next upgrade of `rusoto_signature` will actually remove the `time` dependency and replace it with `chrono`

Comment (syphar) [reply Cargo.lock:4038]: > and as I understand it we really can't get rid of 0.1 till migrating off `iron`

not only `iron` (or `hyper` in that case). See above, in `zip` and `chrono`.

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**